src/Lecture2.hs

```diff
@@ -158,13 +182,62 @@ You're free to define any helper functions.
 -}
 
 -- some help in the beginning ;)
+type Health = Int
+type Endurance = Int
+type Power = Int
+
 data Knight = Knight
-    { knightHealth    :: Int
-    , knightAttack    :: Int
-    , knightEndurance :: Int
+    { knightHealth    :: Health
+    , knightAttack    :: Power
+    , knightEndurance :: Endurance
     }
 
-dragonFight = error "TODO"
+data Dragon = Dragon
+  { dragonType :: Int
+  , dragonHealth :: Health
+  , dragonAttack :: Power
+  , dragonAcid :: Bool
+  , dragonChest :: Chest
+  , dragonHitsTaken :: Int
+  }
+
+data Chest = Chest
+  { chestGold :: Int
+  , chestTreasure :: Int
+  }
+
+data DragonType
```

Comment:
You can export `Dragon` and `DragonType` data types from this module as well in the beginning. Otherwise, it's a bit hard to review due to GHC warnings 😞 

You can add lines like this in the module header:

```haskell
    , Dragon (..)
    , DragonType (..)
```